File: src/main/kotlin/dev/jefy/connectpro/management/domain/repositoty/BadgeRepository.java
package dev.jefy.connectpro.management.domain.repositoty;

import org.springframework.stereotype.Repository;

import dev.jefy.connectpro.management.domain.Badge;
import dev.jefy.connectpro.management.domain.vo.BadgeId;
import dev.jefy.connectpro.shared.infrastructure.ddd.AggregateRepository;

/**
 * @author Jôph Yamba
 */
@Repository
public interface BadgeRepository extends AggregateRepository<Badge, BadgeId> {
    boolean existsByName(String name);
}

File: src/main/kotlin/dev/jefy/connectpro/management/domain/repository/BadgeRepository.kt
package dev.jefy.connectpro.management.domain.repository

import org.springframework.stereotype.Repository
import dev.jefy.connectpro.management.domain.Badge
import dev.jefy.connectpro.management.domain.vo.BadgeId
import org.springframework.data.jpa.repository.JpaRepository

@Repository
interface BadgeRepository : JpaRepository<Badge, BadgeId> {
    fun existsByName(name: String): Boolean
}